src/app/shared/components/batchupload/batchupload.component.ts

```diff
@@ -0,0 +1,370 @@
+    this.data = (XLSX.utils.sheet_to_json(ws, {header : 1}));        // Convert data to json
+    this.createTable(this.data);
+    this.tableDisplay = true;
+    this.sheetNamesButtons = false;
+  }
+
+  async createTable(data) {
+    this.success = [];
+    this.clicked = false;
+    var counter = 0;
+    var explanatoryVariablesArray: explanatoryVariables[] = [];
+    var tempExplanatoryVariablesArray: explanatoryVariables[] = [];
+    var errorsArray:errors[] = [];
+    var explanatoryVariables: explanatoryVariables = {  
+      code: null,
+      limits: ({
+          max: null,
+          min: null,
+      }),
+      unitType: { 
+        id: null 
+      }
+    };
+    var errors: errors = {  
+      id: null,
+      value:  null 
+    };
+
+    // Loop through spreadsheet
+    for (var i = 2; i < data.length; i++) { 
+      if (data[i][0]) { // Study area
+        var studyArea = (data[i][0]);
+      } if (data[i][2]) { // Statistic Group
+        var statisticGroup = (data[i][2]);
+      } if (data[i][5]) { // Regression Region
+        var regressionRegionName = (data[i][5]);
+        const regionID = this.regions.find(r => r.name == studyArea).id;
+        this.regressionRegions = await this._settingsservice.getEntities(this.configSettings.nssBaseURL + 'regressionregions?regions=' + regionID).toPromise();
+        var regressionRegionCode = this.regressionRegions.find(rr => rr.name == regressionRegionName).code;
+        var regressionRegionID = this.regressionRegions.find(rr => rr.name == regressionRegionName).id;
+      }
+      if (data[i][15]) {  // Regression (statistic) variable
+        var regressionVariable = (data[i][15]);
+      }
+      if (data[i][16]) {  // Unit type
+        var unitType = (data[i][16]);
+      }
+      if (data[i][11]) {  // Explanatory Variables
+        explanatoryVariables.code = data[i][11];
+        explanatoryVariables.limits.min = data[i][12];
+        explanatoryVariables.limits.max = data[i][13];
+        explanatoryVariables.unitType.id  = this.unitTypes.find(ut => ut.abbreviation == (data[i][14])).id;
+        explanatoryVariablesArray.push(JSON.parse(JSON.stringify(explanatoryVariables)));
+        tempExplanatoryVariablesArray = explanatoryVariablesArray;
+      }
+      // Errors
+      if (data[i][17]) { // 	Average standard error (of either estimate or prediction)
+        errors.id = this.errors.find(et => et.code == ("SE")).id;
+        errors.value = data[i][17];
+        errorsArray.push(JSON.parse(JSON.stringify(errors)));
+      }
+      if (data[i][19]) { // Average standard error of prediction
+        errors.id = this.errors.find(et => et.code == ("ASEp")).id;
+        errors.value = data[i][19];
+        errorsArray.push(JSON.parse(JSON.stringify(errors)));
+      }
+      if (data[i][28]) { // Percent Correct
+        errors.id = this.errors.find(et => et.code == ("PC")).id;
+        errors.value = data[i][28];
+        errorsArray.push(JSON.parse(JSON.stringify(errors)));
```

Comment:
Are we able to get the column name from the spreadsheet? there's a possibility of new errors being added, or the codes being changed. I think instead of getting a specific column number, we should use the column name to find the right code. This might mean that we need to change the column names to _just_ be the abbreviation, and not include the name (e.g. change "SE (Standard Error)" to just say "SE", then we can loop through the error types list from our services and look for that in the current row. We also might want to think about grabbing column names for other items, in case we add additional columns in the future